coreos-virt-install

```diff
@@ -0,0 +1,92 @@
+#!/usr/bin/python3
+# Wrapper for virt-install designed to run in a container
+# environment, and using defaults targeted for CoreOS.
+# https://github.com/cgwalters/coreos-assembler/issues/7
+# Today, this requires a Kickstart (and an installer ISO path);
+# we might change things down the line to have defaults for
+# one or both.
+
+import os,sys,argparse,subprocess,io,time,re,multiprocessing
+
+def fatal(msg):
+    print('error: {}'.format(msg), file=sys.stderr)
+    raise SystemExit(1)
+
+if os.getuid() == 0:
+    fatal("This script should be run as non-root in a container")
+
+parser = argparse.ArgumentParser()
+parser.add_argument("--dest", help="Destination disk",
+                    action='store', required=True)
+parser.add_argument("--create-disk", help="Automatically create disk as qcow2, parsing kickstart for size",
+                    action='store_true')
+parser.add_argument("--kickstart", help="Kickstart path",
+                    action='store', required=True)
+parser.add_argument("--location", help="Installer location",
+                    action='store', required=True)
+parser.add_argument("--memory", help="Memory size in MB",
+                    action='store', type=int, default=2048)
+parser.add_argument("--console-log-file", help="Console log file",
+                    action='store')
+parser.add_argument("--wait", help="Number of minutes to wait",
+                    action='store', type=int, default=10)
+args = parser.parse_args()
+
+# Copy of bits from https://pagure.io/standard-test-roles/pull-request/223
+def get_libvirt_smp_arg():
+    """Determine the number of CPUs that should be visible in the guest.
+    See e.g. https://www.redhat.com/archives/libvirt-users/2017-June/msg00025.html
+    We want to match the number of host physical cores.
+    """
+    # We may be run in a cgroup with fewer cores available than physical.
+    available_cpu = int(subprocess.check_output(['nproc']).strip())
+    # https://stackoverflow.com/questions/6481005/how-to-obtain-the-number-of-cpus-cores-in-linux-from-the-command-line
+    core_sockets = set()
+    for line in io.BytesIO(subprocess.check_output(['lscpu', '-b', '-p=Core,Socket'])):
+        if line.startswith(b'#'):
+            continue
+        core_sockets.add(line.strip())
+    sockets = min(available_cpu, len(core_sockets))
+    return '--vcpus=sockets={},cores=1,threads=1'.format(sockets, sockets)
+
+def run_sync_verbose(argv, **kwargs):
+    print("cmd: {}".format(subprocess.list2cmdline(argv)))
+    subprocess.check_call(argv, **kwargs)
+
+# This is an "extension" to kickstart files that we implement as a comment.
+if args.create_disk:
+    magic_virt_install_size_str = '#--virt-install-disk-size-gb:'
+    disk_re = re.compile(r'^' + magic_virt_install_size_str + ' *([0-9]+)$')
```

Comment:
That's a lot of magic :unicorn:. Why not just make this an argument to the script itself? Making it explicit would make it much clearer.

Reply:
But that just raises the question of where the configuration for that cmdline argument lives.  Today it's in the TDL which I find obscure.  My rationale here is that the final disk size is *strongly* related to the partitioning in the kickstart, so they should live together.

Reply:
To be clear though, `coreos-virt-install` does also support a mode where the disk is pre-initialized by the caller, which allows full control (generate a raw image, generate a qcow2 with whatever size/format) to the caller.

Reply:
>  My rationale here is that the final disk size is strongly related to the partitioning in the kickstart, so they should live together.

Ahh OK, that's a convincing point. Though can we prefix with `coreos-` too? I.e.. `coreos-virt-install-...`.